contracts/vaults/TreasuryVault/RibbonTreasuryVault.sol

```diff
@@ -0,0 +1,1196 @@
+        );
+
+        emit PerformanceFeeSet(performanceFee, newPerformanceFee);
+
+        performanceFee = newPerformanceFee;
+    }
+
+    /**
+     * @notice Sets a new cap for deposits
+     * @param newCap is the new cap for deposits
+     */
+    function setCap(uint256 newCap) external onlyOwner {
+        require(newCap > 0, "!newCap");
+        ShareMath.assertUint104(newCap);
+        emit CapSet(vaultParams.cap, newCap);
+        vaultParams.cap = uint104(newCap);
+    }
+
+    /**
+     * @notice Sets the new discount on premiums for options we are selling
+     * @param newPremiumDiscount is the premium discount
+     */
+    function setPremiumDiscount(uint256 newPremiumDiscount) external onlyOwner {
+        require(
+            newPremiumDiscount > 0 &&
+                newPremiumDiscount < 100 * Vault.PREMIUM_DISCOUNT_MULTIPLIER,
+            "Invalid discount"
+        );
+
+        emit PremiumDiscountSet(premiumDiscount, newPremiumDiscount);
+
+        premiumDiscount = newPremiumDiscount;
+    }
+
+    /**
+     * @notice Sets the new auction duration
+     * @param newAuctionDuration is the auction duration
+     */
+    function setAuctionDuration(uint256 newAuctionDuration) external onlyOwner {
+        require(
+            newAuctionDuration >= MIN_AUCTION_DURATION,
+            "Invalid auction duration"
+        );
+
+        emit AuctionDurationSet(auctionDuration, newAuctionDuration);
+
+        auctionDuration = newAuctionDuration;
+    }
+
+    /**
+     * @notice Sets the new strike selection contract
+     * @param newStrikeSelection is the address of the new strike selection contract
+     */
+    function setStrikeSelection(address newStrikeSelection) external onlyOwner {
+        require(newStrikeSelection != address(0), "!newStrikeSelection");
+        strikeSelection = newStrikeSelection;
+    }
+
+    /**
+     * @notice Sets the new options premium pricer contract
+     * @param newOptionsPremiumPricer is the address of the new strike selection contract
+     */
+    function setOptionsPremiumPricer(address newOptionsPremiumPricer)
+        external
+        onlyOwner
+    {
+        require(
+            newOptionsPremiumPricer != address(0),
+            "!newOptionsPremiumPricer"
+        );
+        optionsPremiumPricer = newOptionsPremiumPricer;
+    }
+
+    /**
+     * @notice Optionality to set strike price manually
+     * @param strikePrice is the strike price of the new oTokens (decimals = 8)
+     */
+    function setStrikePrice(uint128 strikePrice)
+        external
+        onlyOwner
+        nonReentrant
+    {
+        require(strikePrice > 0, "!strikePrice");
+        overriddenStrikePrice = strikePrice;
+        lastStrikeOverrideRound = vaultState.round;
+    }
+
+    /**
+     * @notice Adds new whitelisted address
+     * @param newWhitelist is a list of address to include to the whitelist
+     */
+    function addWhitelist(address[] calldata newWhitelist)
```

Comment:
I really think we could just simplify `addWhitelist` and `removeWhitelist` to just take in a single address as a parameter. Since it's called by the `owner`, we can batch multiple `addWhitelist` calls into a single transaction

The simpler the better!